src/main/java/nextstep/courses/domain/PaidSession.java

```diff
@@ -0,0 +1,39 @@
+package nextstep.courses.domain;
+
+import nextstep.payments.domain.Payment;
+
+public class PaidSession extends Session {
+    private Capacity maxCapacity;
+    private TuitionFee tuitionFee;
+
+    public PaidSession(Long id, String name, Period period, Image coverImage,
+                       SessionStatus status, int maxCapacity, int tuitionFee) {
+        super(id, name, period, coverImage, status);
+        this.maxCapacity = new Capacity(maxCapacity);
+        this.tuitionFee = new TuitionFee(tuitionFee);
+    }
+
+    @Override
+    protected void validateRegistration(Long studentId, Payment payment) {
+        if (registeredStudents.size() >= maxCapacity.getValue()) {
+            throw new IllegalStateException("수강 인원을 초과하였습니다.");
+        }
+        if (payment == null) {
+            throw new IllegalArgumentException("결제 정보가 없습니다.");
+        }
+        if (!this.id.equals(payment.getSessionId())) {
+            throw new IllegalArgumentException("결제한 강의와 일치하지 않습니다.");
+        }
+        if (!studentId.equals(payment.getNsUserId())) {
+            throw new IllegalArgumentException("결제한 사용자와 일치하지 않습니다.");
+        }
+        if (this.tuitionFee.getAmount() != payment.getAmount()) {
```

Comment:
게터 사용대신 tuitionFee에게 메시지를 보내보면 어떨까요?

메소드 네이밍은 참고만 해주세요~
```suggestion
        if (this.tuitionFee.isSameAmount(payment.getAmount())) {
```